### PR TITLE
chore: change SDK ClientRequest to be interface

### PR DESCRIPTION
**Pull request by adriantam**
## Description
Wrap some of the SDK client request (such as SdkClientReadAssertionsRequest) to have an interface version.  This allows better ability to test and mock by users of the SDK

## References
Close https://github.com/openfga/go-sdk/issues/23

## Review Checklist
- [X] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [X] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -483,106 +508,152 @@ type ClientCreateStoreOptions struct {
...
-	Client OpenFgaClient
+	Client *OpenFgaClient
 
 	options *ClientDeleteStoreOptions
 }
+
+type SdkClientDeleteStoreRequestInterface interface{
+	Options(options ClientDeleteStoreOptions) SdkClientDeleteStoreRequestInterface
+	Execute() (*ClientDeleteStoreResponse, error)	
+
```
Comment (rhamzeh) [inline client/client.go:618]: Something off with the formatting here, can you run gofmt?

File: client/client.go [hunk 2]
```diff
@@ -630,22 +717,33 @@ func (client *OpenFgaClient) ReadAuthorizationModelsExecute(request SdkClientRea
...
 	ctx    _context.Context
-	Client OpenFgaClient
+	Client *OpenFgaClient
 
 	body    *ClientWriteAuthorizationModelRequest
 	options *ClientWriteAuthorizationModelOptions
 }
 
+type SdkClientWriteAuthorizationModelRequestInterface interface {
+    Options(options ClientWriteAuthorizationModelOptions) SdkClientWriteAuthorizationModelRequestInterface
```
Comment (rhamzeh) [inline client/client.go:737]: Formatting

**Review by rhamzeh — CHANGES_REQUESTED** [review]

File: client/client.go (hunk 1 above)
Comment (rhamzeh) [reply client/client.go:618]: Same with several sections below

Comment (adriantam) [reply client/client.go:618]: Did not bring in https://github.com/openfga/sdk-generator/pull/131

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**